src/main/java/org/candlepin/subscriptions/tally/facts/FactNormalizer.java

```diff
@@ -329,6 +330,22 @@ private void handleSla(
   private void normalizeQpcFacts(NormalizedFacts normalizedFacts, InventoryHostFacts hostFacts) {
     // Check if this is a RHEL host and set product.
     if (hostFacts.getQpcProducts() != null && hostFacts.getQpcProducts().contains("RHEL")) {
+      if (hostFacts.getSystemProfileArch() != null
+          && CollectionUtils.isEmpty(hostFacts.getQpcProductIds())) {
```

Comment:
Since https://github.com/RedHatInsights/rhsm-subscriptions/pull/2604, qpcProductIds no longer exists. so I think it's safe to remove this condition.

Reply:
Thanks, I actually think this should have been SystemProfileProductIds to begin with so I updated to check for that, same for the test.